src/Microsoft.Azure.WebJobs.Host/Bindings/FunctionBindingContext.cs

```diff
@@ -89,7 +89,7 @@ public FunctionBindingContext(IFunctionInstanceEx functionInstance, Cancellation
         /// <summary>
         /// The service provider for the current function invocation scope.
         /// </summary>
-        public IServiceProvider InstanceServices { get; set; }
+        public IServiceProvider InstanceServices => this._functionInvocationServices;
```

Comment:
Hmmm, looks like this issue has existed since this property was originally added in https://github.com/Azure/azure-webjobs-sdk/commit/f7da27d133b9ce0aadf9918770f6ed45748f3c8a.

Please add a test to the existing FunctionBindingContext test suite that tests this property and potentially others as well.